src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs

```diff
@@ -126,9 +127,7 @@ public async Task TestPreviewServices()
             var service = previewWorkspace.Services.GetService<ISolutionCrawlerRegistrationService>();
             Assert.IsType<PreviewSolutionCrawlerRegistrationServiceFactory.Service>(service);
 
-            var persistentService = previewWorkspace.Services.GetRequiredService<IPersistentStorageService>();
-
-            await using var storage = await persistentService.GetStorageAsync(previewWorkspace.CurrentSolution, CancellationToken.None);
+            await using var storage = await previewWorkspace.CurrentSolution.GetPersistentStorageAsync(checkBranchId: true, CancellationToken.None);
             Assert.IsType<NoOpPersistentStorage>(storage);
```

Comment:
instead of a two-stage acquisition mechanism, i added a simpel extension for consumers that are dealing with a real Solution snapshot.  This snapshot is used to get the HostServices, Options, and SolutionKey that form teh underlying raw API for storage.

Reply:
I heavily prefer the extension method be a separate PR. It's making this pull request very difficult to review.